pkgerrors/stacktrace.go

```diff
@@ -0,0 +1,66 @@
+package pkgerrors
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/internal/json"
+)
+
+var (
+	StackSourceFileName     = "source"
+	StackSourceLineName     = "line"
+	StackSourceFunctionName = "func"
+)
+
+// MarshalStack implements pkg/errors stack trace marshaling.
+//
+//   zerolog.ErrorStackMarshaler = MarshalStack
+func MarshalStack(err error) []byte {
+	type stackTracer interface {
+		StackTrace() errors.StackTrace
+	}
+	var st errors.StackTrace
+	if err, ok := err.(stackTracer); ok {
```

Comment:
You can remove this `else` by inverting the validation logic.

```go
err, ok := err.(stackTracer)
if !ok {
    return nil
}
return appendJSONStack(make([]byte, 0, 500), err.StackTrace())
```

Just a different point of view on this block 😃